jsonexample/test/simple_object_unit_test.dart

```diff
@@ -0,0 +1,196 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:jsonexample/built_value/built_simple_object.dart';
+import 'package:jsonexample/built_value/built_value_serializers.dart';
+import 'package:jsonexample/dart_convert/converted_simple_object.dart';
+import 'package:jsonexample/json_serializable/serializable_simple_object.dart';
+
+void main() {
+  const Map<String, dynamic> typicalObjectJson = {
+    'aString': 'Blah, blah, blah.',
+    'anInt': 1,
+    'aDouble': 1.0,
+    'aListOfStrings': ['one', 'two', 'three'],
+    'aListOfInts': [1, 2, 3],
+    'aListOfDoubles': [1.0, 2.0, 3.0]
+  };
+
+  const Map<String, dynamic> emptyListJson = {
+    'aString': 'Blah, blah, blah.',
+    'anInt': 1,
+    'aDouble': 1.0,
+    'aListOfStrings': [],
+    'aListOfInts': [],
+    'aListOfDoubles': []
+  };
+
+  const Map<String, dynamic> unexpectedPropertyjson = {
+    'aString': 'Blah, blah, blah.',
+    'anInt': 1,
+    'aDouble': 1.0,
+    'aListOfStrings': ['one', 'two', 'three'],
+    'aListOfInts': [1, 2, 3],
+    'aListOfDoubles': [1.0, 2.0, 3.0],
+    'unexpectedProperty': 'Whoops!'
+  };
+
+  group('ConvertedSimpleObject unit tests', () {
+    test('Typical object', () {
+      final simpleObject = ConvertedSimpleObject.fromJson(typicalObjectJson);
+
+      expect(simpleObject, isNotNull);
```

Comment:
[nit] Purists would say that there should be one expect statement per unit test. I'm not a purist.

I do think these tests may be unnecessarily thorough. Especially below, where you're using json_serializable. Those tests look like something I could find in the json_serializable package. Here we could just make sure that the things don't throw, or that one of the members is what we expect it to be.

Definitely a nit, though.

Reply:
You're right, but I'm not sure what other tests would go in here beyond the type of thing I have. I do like that it effectively compares all three approaches to ensure that they all pass the same tests, which is nice.